### PR TITLE
Use internal sbt dependency resolution in codegen-plugin

### PR DESCRIPTION
**Pull request by schmeedy**
Fixes dependency resolution from a maven repository that requires authentication (tested against a CodeArtifact repo with `sbt-codeartifact`).

### CODE REVIEW

**Comment by schmeedy** [comment]
👍 that sounds better. I gave it a shot (and updated this PR). I'm not quite sure if I found the best way to resolve the dependencies in sbt, but it seems to work OK.

File: modules/codegen-cli/src/smithy4s/codegen/cli/CodegenCommand.scala [hunk 1]
```diff
@@ -125,7 +125,8 @@ object CodegenCommand {
             excludedNS,
             repositories.getOrElse(List.empty),
             dependencies.getOrElse(List.empty),
-            transformers.getOrElse(List.empty)
+            transformers.getOrElse(List.empty),
+            localJars = List.empty
```
Comment (Baccata) [inline modules/codegen-cli/src/smithy4s/codegen/cli/CodegenCommand.scala:129]: Could we actually make that a CLI argument, for completeness. You can follow the exact same patterns as the other ones that have `getOrElse(List.empty)` above this one

File: modules/codegen-plugin/src/smithy4s/codegen/CodegenPlugin.scala [hunk 2]
```diff
@@ -108,15 +108,19 @@ object Smithy4sCodegenPlugin extends AutoPlugin {
...
-        if (CrossVersion.disabled == m.crossVersion)
-          s"${m.organization}:${m.name}:${m.revision}"
-        else s"${m.organization}::${m.name}:${m.revision}"
-      }
-      .toList
+  private def findCodeGenDependencies(
+      updateReport: UpdateReport
+  ): List[os.Path] =
+    for {
+      smithy4sConfigReport <- updateReport.configurations
```
Comment (Baccata) [inline modules/codegen-plugin/src/smithy4s/codegen/CodegenPlugin.scala:115]: @daddykotex this looks good to me, but I'd like your eyes on this.

File: modules/codegen/src/smithy4s/codegen/CodegenArgs.scala [hunk 3]
```diff
@@ -27,5 +27,6 @@ case class CodegenArgs(
     excludedNS: Option[Set[String]],
     repositories: List[String],
     dependencies: List[String],
-    transformers: List[String]
+    transformers: List[String],
+    localJars: List[os.Path]
```
Comment (Baccata) [inline modules/codegen/src/smithy4s/codegen/CodegenArgs.scala:31]: 👍

**Comment by schmeedy** [comment]
Thanks guys, I've added `localJars` as a CLI option.

File: modules/codegen-cli/src/smithy4s/codegen/cli/Options.scala [hunk 4]
```diff
@@ -40,6 +40,17 @@ object Options {
 
   }
 
+  implicit def commaDelimitedListArg[A: Argument]: Argument[List[A]] = {
```
Comment (Baccata) [inline modules/codegen-cli/src/smithy4s/codegen/cli/Options.scala:43]: 👍

**Comment by Baccata** [comment]
This is great, thank you ! We're aiming to a next release this week or the next 😄

**Review by Baccata — APPROVED** [review]

**Pull request merged**